src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3152,6 +3153,313 @@ void MacroAssembler::check_klass_subtype(Register sub_klass,
   BLOCK_COMMENT("} check_klass_subtype");
 }
 
+// scans r_count pointer sized words at [r_addr] for occurrence of r_value,
+// generic (r_count must be >0)
+// iff found: CC eq, r_result == 0
+void MacroAssembler::repne_scan(Register r_addr, Register r_value, Register r_count, Register r_result) {
+  NearLabel L_loop, L_exit;
+
+  BLOCK_COMMENT("repne_scan {");
+#ifdef ASSERT
+  z_chi(r_count, 0);
+  asm_assert(bcondHigh, "count must be positive", 11);
+#endif
+
+  clear_reg(r_result, true /* whole_reg */, false /* set_cc */);  // sets r_result=0, let's hope that search will be successful
+
+  bind(L_loop);
+  z_cg(r_value, Address(r_addr));
+  z_bre(L_exit); // branch on success
+  add2reg(r_addr, wordSize);
```

Comment:
Can add2reg change CC? I'd probably rather use la / lay for the addition without changing CC.

Reply:
I just noticed that `PreferLAoverADD` is by default set to false. I'm looking for a reason for that, maybe we can set it to true. Otherwise I'll update it to z_la. 

CC: @RealLucy

Reply:
Use LA explicitly here.

Reply:
done;